src/query/api/v1/handler/namespace/common_test.go

```diff
@@ -138,6 +141,47 @@ func TestMetadata(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+type testExtendedOptions struct {
```

Comment:
nit: this is defined a couple of times; maybe move it to `/x/test`? If we go ahead with the proposed refactor, it could just return a test implementation of `ExtendedOptionsConverterRegistry`, or better yet just use a mock for these tests

Reply:
Good point. It even has some test protobuf messages that I can take advantage of.